### PR TITLE
Add gradient accumulation for LLC estimation

### PR DESCRIPTION
**Pull request by georgeyw**

### CODE REVIEW

**Comment by georgeyw** [comment]
leaving a note as a reminder that this might also require changing the automatic temp setting in LLC estimation

File: src/devinterp/slt/sampler.py [hunk 1]
```diff
@@ -38,61 +38,93 @@ def sample_single_chain(
...
+            loss.backward()
+
+            # i+1 instead of i so that the gradient accumulates to an entire batch first
+            # otherwise the first draw happens after batch_size/grad_accum_steps samples instead of batch_size samples
+            if (i+1) % grad_accum_steps == 0:
+                optimizer.step()
+                optimizer.zero_grad()
+                pbar.update(1)
+
+            if i >= num_burnin_steps and (i + 1 - num_burnin_steps) % num_steps_bw_draws == 0:
```
[the remark this answers is not in the document]
Comment (georgeyw) [reply src/devinterp/slt/sampler.py:117]: good question, but i think no, at line 55 we have
```
if grad_accum_steps > 1:
        assert type(grad_accum_steps) == int, "grad_accum_steps must be an integer."
        num_steps_bw_draws *= grad_accum_steps
        num_burnin_steps *= grad_accum_steps
```
so this `if` block only triggers once the gradient accumulation is "full" -- the condition for when it happens is identical the original one, but scaled by # of gradient accumulation steps

another sanity check is that loss traces would look like clusters with step changes between them, especially during the period of convergence, which iirc they didn't look like. but if/when we use grad accumulation again, we'll be able to notice at that time as well
